Settings.c

```diff
@@ -847,8 +849,9 @@ Settings* Settings_new(unsigned int initialCpuCount, Hashtable* dynamicMeters, H
             unlink(legacyDotfile);
          }
       }
-      free(legacyDotfile);
    }
+   free(legacyDotfile);
+
```

Comment:
Moving it down to the end of the function in my PR was intentional (to avoid splitting the various steps for reading the config with this cascade of `if (!ok …)` statements …